### DISCUSSION TITLE
Special token for fine-tuning ESM1b

### DISCUSSION THREAD

**Discussion by vuhongai** [facebookresearch/esm#247] [Q&A]
Dear authours,

I would like to use ESM-1b for finetuning a downstream task. The inputs are 2 distant regions of a protein, for example ''M M F S" and "Y M V V". I realize that the model vocab doesn't contain [SEP] token as in orginal BERT, so I can't do like other BERT model "M M F S [SEP] Y M V V". What would you recommend the appropreate input for such task, eos or unk or mask should be used?

Thank you.
Ai

**Comment by tomsercu** [comment]
First let's note that this kind of input is out of distribution for the model's training data.
But there are a few ways that might work, for example
* adding a bunch of mask tokens in the middle
* adding a poly-glycine linker and maybe hack a jump in residue index into the positional embeddings. This is done in [ESMFold](https://www.biorxiv.org/content/10.1101/2022.07.20.500902v1) (quote below) as well as in colabfold and others

> To predict complexes shown in Figure 2D, we give a residue index break of 1000 to ESMFold and link chains with a 25-residue poly-glycine linker, which we remove before displaying. Note that this is using ESMFold out of distribution since single chains are used during training.
